storm/ia32/wrapper.c

```diff
@@ -1,1080 +1,978 @@
+void wrapper_kernelfs_entry_read(void)
 {
   asm ("pushal\n"
-                /* Push all arguments. This is pretty smart... */
-
-                "pushl  32 + 4 + 1 * 4(%esp)\n"
 
-                "call	system_call_kernelfs_entry_read\n"
+                // Push all arguments. This approach pretty smart; it utilizes the fact that the stack grows downwards
+                // so the "next parameter to push" is always in the same memory location. :)
+      
+                "pushl 32 + 4 + 1 * 4(%esp)\n"
+                "call   system_call_kernelfs_entry_read\n"
 
-                "addl	$4 * 1, %esp\n"
+                "addl   $4 * 1, %esp\n"
 
-		/* Simulate a popa, without overwriting EAX. */
+                // Simulate a popa, without overwriting EAX (since it contains the return value from the system call).
+                "popl   %edi\n"
+                "popl   %esi\n"
+                "popl   %ebp\n"
 
-		"popl	%edi\n"
-		"popl	%esi\n"
-		"popl	%ebp\n"
+                // ESP can't be popped for obvious reasons.
+                "addl   $4, %esp\n"
+                "popl   %ebx\n"
+                "popl   %edx\n"
+                "popl   %ecx\n"
 
-		/* ESP can't be popped for obvious reasons. */
-
-		"addl	$4, %esp\n"
-		"popl	%ebx\n"
-		"popl	%edx\n"
-		"popl	%ecx\n"
-
-		/* EAX shall not be changed, since it is our return
-          	  value. */
-
-		"addl	$4, %esp\n"
-		"lret	$4 * 1\n");
+                // Adjust the stack for the fact that EAX isn't being popped.
+                "addl   $4, %esp\n"
+                "lret   $4 * 1\n");
 }
 
-void wrapper_mailbox_create (void)
+void wrapper_mailbox_create(void)
 {
   asm ("pushal\n"
-                /* Push all arguments. This is pretty smart... */
-
-                "pushl  32 + 4 + 5 * 4(%esp)\n"
-                "pushl  32 + 4 + 5 * 4(%esp)\n"
-                "pushl  32 + 4 + 5 * 4(%esp)\n"
-                "pushl  32 + 4 + 5 * 4(%esp)\n"
-                "pushl  32 + 4 + 5 * 4(%esp)\n"
 
-                "call	system_call_mailbox_create\n"
+                // Push all arguments. This approach pretty smart; it utilizes the fact that the stack grows downwards
+                // so the "next parameter to push" is always in the same memory location. :)
+      
+                "pushl 32 + 4 + 5 * 4(%esp)\n"
+                "pushl 32 + 4 + 5 * 4(%esp)\n"
+                "pushl 32 + 4 + 5 * 4(%esp)\n"
+                "pushl 32 + 4 + 5 * 4(%esp)\n"
+                "pushl 32 + 4 + 5 * 4(%esp)\n"
+                "call   system_call_mailbox_create\n"
 
-                "addl	$4 * 5, %esp\n"
+                "addl   $4 * 5, %esp\n"
 
-		/* Simulate a popa, without overwriting EAX. */
+                // Simulate a popa, without overwriting EAX (since it contains the return value from the system call).
+                "popl   %edi\n"
+                "popl   %esi\n"
+                "popl   %ebp\n"
 
-		"popl	%edi\n"
-		"popl	%esi\n"
-		"popl	%ebp\n"
```

Comment:
The indentation was "off by one" in the output that the old script used to create, so I fixed that while I was at it.